### PR TITLE
Add expense admin action buttons to mobile view on expense summary

### PR DESCRIPTION
**Pull request by sbinlondon**
Fixes https://github.com/opencollective/opencollective/issues/3785

Screenshot 2021-01-25 at 18 05 59

### CODE REVIEW

**Review by Betree — COMMENTED** [review]
Code looks perfect, buttons work.

- (not a blocker) on desktop, we probably want to use the full width for the expense title here:
image


- From what @Memo-Es said on Slack, I think we also want to implement these buttons on the expense page, mobile version (`/collective/expenses/:id`). They're still missing in there.

**Comment by sbinlondon** [comment]
>     * From what @Memo-Es said on Slack, I think we also want to implement these buttons on the expense page, mobile version (`/collective/expenses/:id`). They're still missing in there.

I misunderstood, I thought y'all had said it was host dashboard only. I'll add it to the main expense page too.

File: components/expenses/ExpenseSummary.js [hunk 1]
```diff
@@ -70,17 +80,38 @@ const ExpenseSummary = ({
...
+                collective={collective}
+                expense={expense}
+                permissions={pick(expense.permissions, ['canSeeInvoiceInfo', 'canDelete'])}
+                buttonProps={{ size: 32, m: 1 }}
+                linkAction="link"
+                onDelete={() => {
+                  onClose();
+                  if (onDelete) {
+                    onDelete(expense);
+                  }
```
Comment (kewitz) [inline components/expenses/ExpenseSummary.js:109]: If `onClose` prop type is not required, we should also rest if it exists before calling it.
There's a shorthand to do that if you're interested:
```suggestion
                  onClose?.();
                  onDelete?.(expense);
```

File: pages/expense.js [hunk 2]
```diff
@@ -500,6 +500,9 @@ class ExpensePage extends React.Component {
                   isLoadingLoggedInUser={loadingLoggedInUser || isRefetchingDataForUser}
                   permissions={expense?.permissions}
                   collective={collective}
+                  onError={error => this.setState({ error })}
```
Comment (kewitz) [inline pages/expense.js:503]: :+1: Nice!

**Review by kewitz — COMMENTED** [review]
Looks good, nice work!
I believe these components are being used across different pages, just make sure these new properties and functions are all optional and we should be gold. :medal_sports:

**Review by kewitz — APPROVED** [review]
LGTM!

**Review by Betree — APPROVED** [review]
Looking good :+1:

**Pull request merged**